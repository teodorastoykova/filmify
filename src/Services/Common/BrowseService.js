import axios from "axios";
import Movie from "../../models/Movie";
import TvSeries from "../../models/TvSeries";

const getBrowseResults = async (searchQuery) => {
  try {
    const response = await axios.get(
      `https://api.themoviedb.org/3/search/multi`,
      {
        params: {
          query: searchQuery,
          include_adult: "false",
          language: "en-US",
        },
        headers: {
          accept: "application/json",
          Authorization:
            "Bearer eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiI1NThkYzk1YTAxNDQyY2NmZDM0YzJlOWY0MmRiNjY5MCIsInN1YiI6IjY2MTNlOGU5MGJiMDc2MDE0ODJmYjYzNSIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.W6UGD_8V7MgsHRv2EizIyg5KV9ZZb9cHGV4A_Nq_UNY",
        },
      }
    );
    const results = response.data.results
    .filter((item) => item.media_type !== 'person')
    .map((item) => {
      if (item.media_type === "movie") {
        return new Movie(
          item.id,
          item.title,
          undefined,
          item.poster_path,
          undefined, 
          undefined, 
          undefined,
          item.media_type
        );
      } else if (item.media_type === "tv") {
        return new TvSeries(
          item.id,
          item.name,
          undefined,
          item.poster_path,
          undefined, 
          undefined, 
          undefined, 
          undefined, 
          undefined,
          item.media_type
        );
      }
    });
    
    return results;
  } catch (error) {
    console.error("Error fetching browse results:", error);
    throw error;
  }
};
export default getBrowseResults;
