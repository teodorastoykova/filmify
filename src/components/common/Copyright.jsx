
import { Typography } from "@mui/material";

const Copyright = (props) => {
    return (
      <Typography
        variant="body2"
        color="text.secondary"
        align="center"
        {...props}
      >
        {"Copyright Â© Filmify "}
        {new Date().getFullYear()}
        {"."}
      </Typography>
    );
  }

export default Copyright;
  