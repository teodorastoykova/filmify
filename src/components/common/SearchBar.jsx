/* eslint-disable react/prop-types */
import { useState } from "react";
import IconButton from "@mui/material/IconButton";
import SearchIcon from "@mui/icons-material/Search";
import NotificationsNoneIcon from "@mui/icons-material/NotificationsNone";
import { TextField, Grid } from "@mui/material";
import { useNavigate } from "react-router";

const SearchBar = () => {
  const [isSearchClicked, setIsSearchClicked] = useState(false);
  const [searchValue, setSearchValue] = useState("");

  const navigate = useNavigate();

  const handleSearchIconClick = () => {
    setIsSearchClicked(!isSearchClicked);
  };

  const handleSearchInputChange = (event) => {
    setSearchValue(event.target.value);
  };

  const handleKeyPress = (event) => {
    if (event.key === 'Enter') {
      goToSearchResults();
    }
  };

  const goToSearchResults = () => {
    if (searchValue.trim() !== '') {
      navigate(`/browse?query=${encodeURIComponent(searchValue)}`);
    }
  };

  return (
    <>
      <Grid item>
        {isSearchClicked && (
          <TextField
            id="outlined-basic"
            label="Search"
            variant="outlined"
            value={searchValue}
            onChange={handleSearchInputChange}
            onKeyPress={handleKeyPress}
            autoFocus
          />
        )}
      </Grid>
      <Grid item>
        <IconButton onClick={handleSearchIconClick}>
          <SearchIcon />
        </IconButton>
        <IconButton>
          <NotificationsNoneIcon />
        </IconButton>
      </Grid>
    </>
  );
};

export default SearchBar;
