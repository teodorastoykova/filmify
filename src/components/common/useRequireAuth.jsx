/* eslint-disable react/prop-types */

const useRequireAuth = () => {
  const sessionId = localStorage.getItem('sessionId');
  return sessionId;
};

export default useRequireAuth;