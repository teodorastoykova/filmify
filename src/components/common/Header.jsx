/* eslint-disable react/prop-types */
import { Button, Grid } from "@mui/material";
import { useNavigate } from "react-router";
import SearchBar from "./SearchBar";

const Header = () => {
  const navigate = useNavigate();

  const handleHomeClick = () => {
    navigate("/");
  };

  const handleTopRatedClick = () => {
    navigate("/top_rated");
  };

  const handleMyWatchListClick = () => {
    navigate("/my_watch_list")
  }
  const handleLogOutClick = () => {
    localStorage.removeItem("sessionId");
    navigate("/login");
  };

  return (
    <>
      <Grid container alignItems="center" justifyContent="space-between">
        <Grid item className="title">
          Filmify
        </Grid>
        <Grid item>
          <Button onClick={handleHomeClick}>Home</Button>
          <Button onClick={handleTopRatedClick}>Top Rated</Button>
          <Button onClick={handleMyWatchListClick}>My Watch List</Button>
          <Button onClick={handleLogOutClick}>Log Out</Button>
        </Grid>
        <SearchBar />
      </Grid>
    </>
  );
};

export default Header;
