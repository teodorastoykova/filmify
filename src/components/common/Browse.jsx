import { useEffect, useState } from "react";
import { useLocation, useNavigate } from "react-router-dom";
import axios from "axios";
import Header from "./Header";
import MoviePoster from "../movies/MoviePoster";
import TvSeriesPoster from "../tvSeries/TvSeriesPoster";
import useRequireAuth from "./useRequireAuth";

const Browse = () => {
  const sessionId = useRequireAuth();
  const navigate = useNavigate();
  const location = useLocation();
  const searchQuery = new URLSearchParams(location.search).get("query");

  const [movies, setMovies] = useState([]);
  const [tvSeries, setTvSeries] = useState([]);
  // eslint-disable-next-line no-unused-vars
  const [searchResults, setSearchResults] = useState([]);

  useEffect(() => {
    if (searchQuery) {
      const options = {
        method: "GET",
        url: "https://api.themoviedb.org/3/search/multi",
        params: {
          query: searchQuery,
          include_adult: "false",
          language: "en-US",
        },
        headers: {
          accept: "application/json",
          Authorization:
            "Bearer eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiI1NThkYzk1YTAxNDQyY2NmZDM0YzJlOWY0MmRiNjY5MCIsInN1YiI6IjY2MTNlOGU5MGJiMDc2MDE0ODJmYjYzNSIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.W6UGD_8V7MgsHRv2EizIyg5KV9ZZb9cHGV4A_Nq_UNY",
        },
      };

      axios
        .request(options)
        .then(function (response) {
          console.log(response.data);

          const results = response.data.results || [];
          setSearchResults(results);

          const moviesList = results.filter(
            (item) => item.media_type === "movie"
          );
          const tvSeriesList = results.filter(
            (item) => item.media_type === "tv"
          );

          setMovies(moviesList);
          setTvSeries(tvSeriesList);
        })
        .catch(function (error) {
          console.error(error);
        });
    }
  }, [searchQuery]);

  const handleMovieClick = (movie) => {
    navigate(`/movies/${movie.id}`);
  };

  const handleTvSeriesClick = (tvSeries) => {
    navigate(`/series/${tvSeries.id}`);
  };
  return sessionId ? (
    <div className="app-container">
      <Header />

      <h1>Search Results For {searchQuery}</h1>
      <h1>Movies</h1>
      <MoviePoster movies={movies} onMovieClick={handleMovieClick} />

      <h1>TV Series</h1>
      <TvSeriesPoster
        tvSeries={tvSeries}
        onTvSeriesClick={handleTvSeriesClick}
      />
    </div>
  ) : null;
};

export default Browse;
