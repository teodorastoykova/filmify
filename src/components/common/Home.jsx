/* eslint-disable react/prop-types */
import "../../App.css";
import { useEffect, useState } from "react";
import axios from "axios";
import Header from "./Header";
import MoviePoster from "../movies/MoviePoster";
import TvSeriesPoster from "../tvSeries/TvSeriesPoster";
import { useNavigate } from "react-router";
import useRequireAuth from "./useRequireAuth";
import { Typography } from "@mui/material";

function Home() {
  const [movies, setMovies] = useState([]);
  const [tvSeries, setTvSeries] = useState([]);

  const navigate = useNavigate();
  const sessionId = useRequireAuth();

  useEffect(() => {
    if (sessionId) {
      const fetchTrendingData = async () => {
        try {
          const trendingMoviesResponse = await axios.get(
            "https://api.themoviedb.org/3/trending/movie/day",
            {
              params: { language: "en-US" },
              headers: {
                accept: "application/json",
                Authorization:
                  "Bearer eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiI1NThkYzk1YTAxNDQyY2NmZDM0YzJlOWY0MmRiNjY5MCIsInN1YiI6IjY2MTNlOGU5MGJiMDc2MDE0ODJmYjYzNSIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.W6UGD_8V7MgsHRv2EizIyg5KV9ZZb9cHGV4A_Nq_UNY",
              },
            }
          );

          const trendingTvResponse = await axios.get(
            "https://api.themoviedb.org/3/trending/tv/day",
            {
              params: { language: "en-US" },
              headers: {
                accept: "application/json",
                Authorization:
                  "Bearer eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiI1NThkYzk1YTAxNDQyY2NmZDM0YzJlOWY0MmRiNjY5MCIsInN1YiI6IjY2MTNlOGU5MGJiMDc2MDE0ODJmYjYzNSIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.W6UGD_8V7MgsHRv2EizIyg5KV9ZZb9cHGV4A_Nq_UNY",
              },
            }
          );
          setMovies(trendingMoviesResponse.data.results);
          setTvSeries(trendingTvResponse.data.results);
        } catch (error) {
          console.error("Error fetching trending data:", error);
        }
      };
      fetchTrendingData();
    } else {
      navigate("/login");
    }
  }, [navigate, sessionId]);

  const handleMovieClick = (movie) => {
    navigate(`/movies/${movie.id}`);
  };

  const handleTvSeriesClick = (tvSeries) => {
    navigate(`/series/${tvSeries.id}`);
  };

  return sessionId ? (
    <>
      <Header />
      <Typography variant="h3" gutterBottom>
        Trending Movies
      </Typography>
      <MoviePoster movies={movies} onMovieClick={handleMovieClick} />
      <Typography variant="h3" gutterBottom>
        Trending TV Series
      </Typography>
      <TvSeriesPoster
        tvSeries={tvSeries}
        onTvSeriesClick={handleTvSeriesClick}
      />
    </>
  ) : null;
}

export default Home;
