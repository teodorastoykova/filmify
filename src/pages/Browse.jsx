import { useEffect, useState } from "react";
import { useLocation, useNavigate } from "react-router-dom";
import Header from "../components/common/Header";
import MoviePoster from "../components/movies/MoviePoster";
import TvSeriesPoster from "../components/tvSeries/TvSeriesPoster";
import useRequireAuth from "../components/common/useRequireAuth";
import { Typography } from "@mui/material";
import getBrowseResults from "../services/Common/BrowseService";

const Browse = () => {
  const sessionId = useRequireAuth();
  const navigate = useNavigate();
  const location = useLocation();
  const searchQuery = new URLSearchParams(location.search).get("query");

  const [movies, setMovies] = useState([]);
  const [tvSeries, setTvSeries] = useState([]);

  useEffect(() => {
    const fetchData = async () => {
      if (searchQuery) {
        try {
          const data = await getBrowseResults(searchQuery);

          const results = data || [];

          const moviesList = results.filter(
            (item) => item.media_type === "movie"
          );
          const tvSeriesList = results.filter(
            (item) => item.media_type === "tv"
          );

          setMovies(moviesList);
          setTvSeries(tvSeriesList);
        } catch (error) {
          console.log("Error fetching data:", error);
        }
      }
    };

    fetchData();
  }, [searchQuery]);

  const handleMovieClick = (movie) => {
    navigate(`/movies/${movie.id}`);
  };

  const handleTvSeriesClick = (tvSeries) => {
    navigate(`/series/${tvSeries.id}`);
  };
  return sessionId ? (
    <>
      <Header />
      <Typography variant="h2" gutterBottom>
        Search Results For {searchQuery}
      </Typography>
      <Typography variant="h3" gutterBottom>
        Movies
      </Typography>
      <MoviePoster movies={movies} onMovieClick={handleMovieClick} />

      <Typography variant="h3" gutterBottom>
        TV Series
      </Typography>
      <TvSeriesPoster
        tvSeries={tvSeries}
        onTvSeriesClick={handleTvSeriesClick}
      />
    </>
  ) : null;
};

export default Browse;
