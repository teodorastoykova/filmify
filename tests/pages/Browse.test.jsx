import { beforeEach, describe, expect, it, vi } from "vitest";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Browse from "../../src/pages/Browse";
import * as getBrowseResults from "../../src/services/Common/BrowseService";
import * as useRequireAuth from "../../src/components/common/useRequireAuth";
import * as router from "react-router";

describe("Browse page", () => {
  const mockGetBrowseResults = vi.fn();
  const mockUseRequireAuth = vi.fn();
  const mockUseNavigate = vi.fn();
  const mockUseLocation = vi.fn();

  const mockedResults = [
    {
      id: 1,
      title: "Movie 1",
      poster_path: "/poster1.jpg",
      media_type: "movie",
    },
    {
      id: 2,
      title: "Movie 2",
      poster_path: "/poster2.jpg",
      media_type: "movie",
    },
    {
      id: 3,
      title: "Movie 3",
      poster_path: "/poster3.jpg",
      media_type: "movie",
    },
    {
      id: 4,
      title: "Movie 4",
      poster_path: "/poster4.jpg",
      media_type: "movie",
    },
    {
      id: 5,
      title: "Movie 5",
      poster_path: "/poster5.jpg",
      media_type: "movie",
    },
    {
      id: 6,
      name: "TV Series 1",
      poster_path: "/poster6.jpg",
      media_type: "tv",
    },
    {
      id: 7,
      name: "TV Series 2",
      poster_path: "/poster7.jpg",
      media_type: "tv",
    },
    {
      id: 8,
      name: "TV Series 3",
      poster_path: "/poster8.jpg",
      media_type: "tv",
    },
    {
      id: 9,
      name: "TV Series 4",
      poster_path: "/poster9.jpg",
      media_type: "tv",
    },
    {
      id: 10,
      name: "TV Series 5",
      poster_path: "/poster10.jpg",
      media_type: "tv",
    },
  ];

  beforeEach(() => {
    vi.spyOn(getBrowseResults, "default").mockImplementation(
      mockGetBrowseResults
    );
    vi.spyOn(useRequireAuth, "default").mockImplementation(mockUseRequireAuth);
  });
  it("renders a snapshot", () => {
    vi.spyOn(router, "useNavigate").mockImplementation(() => mockUseNavigate);
    vi.spyOn(router, "useLocation").mockImplementation(() => mockUseLocation);
    
    mockGetBrowseResults.mockResolvedValue(mockedResults);
    mockUseRequireAuth.mockResolvedValue(1);
    mockUseLocation.mockResolvedValue({ search: "?query=test" });
    const { container } = render(
      <MemoryRouter>
        <Browse />
      </MemoryRouter>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
