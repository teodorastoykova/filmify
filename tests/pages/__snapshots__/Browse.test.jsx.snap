// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Browse page > renders a snapshot 1`] = `null`;
